Sources/XcodeGraph/Models/XCFrameworkInfoPlist.swift

```diff
@@ -9,6 +9,14 @@ public struct XCFrameworkInfoPlist: Codable, Hashable, Equatable {
 
     /// It represents a library inside an .xcframework
     public struct Library: Codable, Hashable, Equatable {
+        public enum Platform: String, CaseIterable, Codable {
```

Comment:
The semantics of this symbol match the semantics of [`Platform`](https://github.com/tuist/XcodeGraph/blob/main/Sources/XcodeGraph/Models/Platform.swift). I'd recommend removing this, and adding a getter to that other symbol that's named `xcframeworkInfoPlistValue: String`.

Reply:
Yes, generally they're match, so I use the [Platform](https://github.com/tuist/XcodeGraph/blob/main/Sources/XcodeGraph/Models/Platform.swift) directly at the beginning. But as @kwridan mentioned at https://github.com/tuist/tuist/pull/6414#pullrequestreview-2150602930

> In this case the issue is that encoded (raw) values of those types are incompatible.
> 
> For example, for visionOS:
> 
> XcodeGraph.Platform: has a raw value of [visionos](https://github.com/tuist/XcodeGraph/blob/main/Sources/XcodeGraph/Models/Platform.swift#L15)
> XCFrameworkInfoPlist has
> 
> <key>SupportedPlatform</key>
> <string>xros</string> 
> As such the Platform type can't be used directly to decode SupportedPlatform and would require an intermediate step to map from the SupportedPlatform to Platform.

There're slightly difference when it comes for `visionOS`. 

> adding a getter to that other symbol that's named `xcframeworkInfoPlistValue: String`.

The value getter is straightforward. However, for `Codable` derived, we must add an `init(xcframeworkInfoPlistValue: String)` and manage the custom decode process for `Library: Codable`. This process is more challenging than custom enum decoding. I suggest transforming it when necessary.

I'm not sure whether I misunderstand, or if there's a better way for custom decoding.

Reply:
I understand the issue now. I think your approach of having a different platform is good. Let's go with it.